### PR TITLE
Game state end game state

### PR DESCRIPTION
**Pull request by Roccubalbetta**

### CODE REVIEW

File: .idea/workspace.xml [hunk 1]
```diff
@@ -0,0 +1,763 @@
...
+              <list>
+                <option value="0.2" />
+              </list>
+            </option>
+          </splitter-proportions>
+        </settings>
+      </state>
+    </states>
+  </component>
+</project>
```
Comment (MyK00L) [inline .idea/workspace.xml:763]: NO

File: .gitignore [hunk 2]
```diff
@@ -8,4 +8,7 @@
 .classpath
 .project
 .classpath
+.project
+.classpath
```
Comment (MyK00L) [inline .gitignore:12]: just how much do you hate .project and .classpath?

File: build.gradle.kts [hunk 3]
```diff
@@ -121,3 +121,19 @@ tasks.spotbugsMain {
...
+    ruleSets.clear()
+    ruleSetFiles(file("$projectDir/config/pmd/pmd.xml"))
+}
+
+tasks.spotbugsMain {
+    reports.create("html") {
+        isEnabled = true
+        setStylesheet("fancy-hist.xsl")
+    }
+}
```
Comment (MyK00L) [inline build.gradle.kts:138]: Why tf are you adding random things that are already there??

**Comment by eutampieri** [comment]
@Roccubalbetta have you fixed what @MyK00L said? Is this PR ready for review?

File: src/main/java/eu/eutampieri/catacombs/model/SimpleWeapon.java [hunk 4]
```diff
@@ -84,7 +84,7 @@ public String getName() {
      * @param delta time between updates
      */
     @Override
-    public void update(final int delta) {
+    public void update(final float delta) {
```
Comment (eutampieri) [inline src/main/java/eu/eutampieri/catacombs/model/SimpleWeapon.java:87]: Can we just use `int`s for time deltas? Floats are bad

**Review by eutampieri — CHANGES_REQUESTED** [review]

File: src/main/java/eu/eutampieri/catacombs/model/SimpleWeapon.java (hunk 4 above)
Comment (Roccubalbetta) [reply src/main/java/eu/eutampieri/catacombs/model/SimpleWeapon.java:87]: Sure

File: src/main/java/eu/eutampieri/catacombs/ui/Game.java [hunk 5]
```diff
@@ -248,10 +248,10 @@ public final void run() {
             timer += now - lastTime;
             updates = 0;
             while ((now - lastUpdateTime) >= tickPerTime) {
-                float delta;
-                delta = (now - lastUpdateTime) / DELTA_CONST;
+                int delta;
+                delta = Math.round((now - lastUpdateTime) / DELTA_CONST);
                 KEY_MANAGER.update(delta);
-                delta = delta <= DELTA_MIN ? delta : DELTA_MIN;
+                delta = Math.round(delta <= DELTA_MIN ? delta : DELTA_MIN);
```
Comment (eutampieri) [inline src/main/java/eu/eutampieri/catacombs/ui/Game.java:254]: Is this needed? Perhaps just make DELTA_MIN an `int`

**Review by eutampieri — COMMENTED** [review]
Please check if those roundings are needed and we'll be good to go!

**Review by jackmounts — APPROVED** [review]

File: src/main/java/eu/eutampieri/catacombs/ui/Game.java (hunk 5 above)
Comment (jackmounts) [reply src/main/java/eu/eutampieri/catacombs/ui/Game.java:254]: math.round might not be needed using integers as well

File: src/main/java/eu/eutampieri/catacombs/ui/EndGameState.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    private final DungeonGame game;
+
+    public EndGameState(final DungeonGame game) {
+        super(game);
+        this.game = game;
+    }
+
+    @Override
+    public void update (final int delta) {
+        this.blinkDelayCount *= delta;
```
Comment (MyK00L) [inline src/main/java/eu/eutampieri/catacombs/ui/EndGameState.java:24]: are you sure about *=, should it not be += ?

Comment (Roccubalbetta) [reply src/main/java/eu/eutampieri/catacombs/ui/EndGameState.java:24]: you are right

Comment (Roccubalbetta) [reply src/main/java/eu/eutampieri/catacombs/ui/EndGameState.java:24]: fixed

**Comment by Roccubalbetta** [comment]
i left float delta

**Review by eutampieri — APPROVED** [review]

**Pull request merged**